vllm/model_executor/model_loader.py

```diff
@@ -15,6 +15,8 @@
     "AquilaModel": AquilaForCausalLM,
     "BaiChuanForCausalLM": BaiChuanForCausalLM,  # baichuan-7b
     "BaichuanForCausalLM": BaichuanForCausalLM,  # baichuan-13b
+    "BaiChuan2ForCausalLM": BaiChuan2ForCausalLM,  # baichuan-7b
```

Comment:
https://huggingface.co/baichuan-inc/Baichuan2-13B-Chat/blob/d022d7264467b2c3bc483e7a3a17105dedba50b8/modeling_baichuan.py#L536

https://huggingface.co/baichuan-inc/Baichuan2-13B-Chat/blob/d022d7264467b2c3bc483e7a3a17105dedba50b8/config.json#L8
According to baichuan2 offical code, they still call their model BaichuanForCausalLM.
Does that mean, if we directly use baichuan2 model download from HF repo, vllm will never load the code for Baichuan2?

Reply:
Thanks for catching this. I pushed a workaround by comparing vocab_size to decide which Baichuan version to call.

in `_get_model_architecture()`
```
# baichuan 2 has different vocab size
if ("baichuan" in arch.lower()) and (getattr(config, "vocab_size")
                                     == 125696):
    return Baichuan2ForCausalLM
```